### ISSUE TITLE
remove components configuration dsl

### ISSUE THREAD

**Issue by lburgazzoli** [apache/camel-k-runtime#160]
in groovy and kotlin dsl we have a dedicated dsl for configuring components and registry entries, like

```groovy
context {
    components {
        seda {
            // set value as method
            queueSize 1234

            // set value as property
            concurrentConsumers = 12
        }

        mySeda(SedaComponent) {
            // set value as method
            queueSize 4321

            // set value as property
            concurrentConsumers = 21
        }
    }

    registry {
        myEntry1 = 'myRegistryEntry1'
        myEntry2 = 'myRegistryEntry2'
    }
}
```

but we have now a `beans` dsl that does almost the same thing like:

```groovy
beans {
    dataSource(org.apache.commons.dbcp2.BasicDataSource) {
        driverClassName = "org.h2.Driver"
        url = "jdbc:h2:mem:camel"
        username = "sa"
        password = ""
    }
}
```

The behavior is very similar so I'd propose to remove support for the `components` and `context` block in favor of the `beans` one.

**Comment by lburgazzoli** [comment]
@davsclaus what do you think ?

**Comment by davsclaus** [comment]
How would a beans look like for configuring those seda options ?

**Comment by lburgazzoli** [comment]
Something like:

```groovy
beans {
    seda(SedaComponent) {
        // set value as method
        queueSize 1234

        // set value as property
        concurrentConsumers = 12
    }
}
```

it is not a compact as the `component` dsl but it is consistent (also for tooling POV)

**Comment by davsclaus** [comment]
Hmm would you then need to FQN the clasname, eg SedaComponent should be org.apache.camel....

And why do you need to set the class name in the first place, when there is seda already. 

Also this would be groovy only? 

And if there is no context block, then how would you set those common options on camel context we have from camel-main etc ?

**Comment by lburgazzoli** [comment]
Maybe better to keep the components block (and add also the dataformat and langiage ones) and remove the registry one in favor of the beans block.

WDYT ?

**Comment by davsclaus** [comment]
Yeah i like to use beans for registry, as that is natural, they are just beans.

The others are camel specific configurations, and allows for copy/paste from the website/docs and from other examples, eg spring boot application.properites etc.

**Comment by davsclaus** [comment]
And btw maybe we can have `context` as `camel` instead ?

```
camel {
   name = dude
   components {
     seda {
       queue-size = 1234
     }
  }
}
```

**Comment by lburgazzoli** [comment]
let me work on that

**Comment by davsclaus** [comment]
btw i like the dash style, not sure if works with that, but camel-main supports this style, so you can say `queueSize` or `queue-size` etc.

**Comment by lburgazzoli** [comment]
I can make it working I think as groovy has some magic, probably not working with kotlin but that's one more reason to use groovy :)

**Comment by lburgazzoli** [comment]
thinking about it a little, `queue-size` wont' be a valid name
